src/cython_code/State.pyx

```diff
@@ -372,18 +372,34 @@ cdef class p_State:
     def insert_row(self, row_data, matching_row_idx, row_idx=-1):
         return self.thisptr.insert_row(row_data, matching_row_idx, row_idx)
     def transition(self, which_transitions=(), n_steps=1,
-                   c=(), r=(), max_iterations=-1, max_time=-1):
+                   c=(), r=(), max_iterations=-1, max_time=-1, progress=None):
+         def _progress(percentage):
+              import sys
+              progress = ' ' * 30
+              fill = int(percentage * len(progress))
+              progress = '[' + '=' * fill + progress[fill:] + ']'
+              print('\r{} {:1.2f}%'.format(progress, 100 * percentage), end="")
+              sys.stdout.flush()
```

Comment:
The diff a few lines of code and the functionality can be disabled with a flag; I am happy to instead scan for a environment variable such as `CROSSCAT_VERBOSE` instead. Either way in this case, the upshot (users can see how far along the inference program has gone, improving usability of the system) is much larger than the downsides (flexibility to hypothetical future applications, or aesthetic notions of what "`transition` should do").

Do you have an alternative user interface design which achieves identical functionality? If yes, please do propose. If no (and might later), then the merge for a feature which frequent users of the system find useful must proceed immediately and cannot be blocked. A ticket can be filed and resolved when the design is clear or it becomes a real problem.